include/nlohmann/detail/exceptions.hpp

```diff
@@ -131,7 +130,20 @@ class exception : public std::exception
         {
             return concat(a, '/', detail::escape(b));
         });
-        return concat('(', str, ") ");
+#if JSON_DIAGNOSTIC_POSITIONS
+        if ((leaf_element->start_pos() != std::string::npos) && (leaf_element->end_pos() != std::string::npos))
```

Comment:
Line 134-139 appears to be similar to 140-146. Should we consider extracting it into a helper function for better code reusability? Like below..

```
template<typename BasicJsonType>
std::string get_byte_positions(const BasicJsonType* leaf_element)
{
#if JSON_DIAGNOSTIC_POSITIONS
    if ((leaf_element->start_pos() != std::string::npos) && (leaf_element->end_pos() != std::string::npos))
    {
        return concat('(', "byte ", std::to_string(leaf_element->start_pos()), "-", std::to_string(leaf_element->end_pos()), ") ");
    }
    return "";
#else
    static_cast<void>(leaf_element);
    return "";
#endif
}
```

Reply:
thanks for the suggestion, yes I will do this, it looks like a good idea to me.